rio/src/RIO/Prelude/Extra.hs

```diff
@@ -75,15 +81,28 @@ nubOrd =
 
 -- | Run the second value if the first value returns 'True'
 whenM :: Monad m => m Bool -> m () -> m ()
-whenM boolM action = do
-  x <- boolM
-  if x then action else return ()
+whenM boolM action = boolM >>= (`when` action)
 
 -- | Run the second value if the first value returns 'False'
 unlessM :: Monad m => m Bool -> m () -> m ()
-unlessM boolM action = do
-  x <- boolM
-  if x then return () else action
+unlessM boolM action = boolM >>= (`unless` action)
+
+
+-- | Same as `(>>)`, but with arguments flipped.
+--
+-- @since 0.1.10.0
+(<<) :: Monad m => m a -> m b -> m a
```

Comment:
Overall not so interested. Shouldn't we be encouraging `*>` and `<*` instead?

Reply:
My sole input on this is that `>>` and `<<` should be a package deal: both, or neither. And I think the arguments presented for "both" are solid.

(I wasn't aware of the associativity difference between `<<` and `<*`... wonder what the reasoning was there?)

Reply:
> First of all they can have different implementation. As such they can have different behavior and at the very least they can have different strictness properties.

Not worried about this, that would be a bug.

> They do restrict the type to a Monad, rather then the Applicative when used in a function.

Why is that relevant?

> Last, but not least, they have different infix precedence, as such *> and <* don't compose well with >>= and =<< respectfully.

Ouch, ok.

Reply:
@tejon Regret to announce, we've been outvoted. `<<` operator is now removed.

Reply:
FTR, this is because I hadn't realized that `<*` and `<<` were intended to behave differently. The motivation makes sense in context of `Monad`, but I'm highly skeptical of creating confusing operator soup at that point.